build.go

```diff
@@ -130,44 +130,42 @@ func main() {
 
 	parseFlags()
 
-	switch goarch {
-	case "386", "amd64", "arm", "arm64", "ppc64", "ppc64le":
-		break
-	default:
-		log.Printf("Unknown goarch %q; proceed with caution!", goarch)
-	}
-
+	checkArchitecture()
 	goVersion, _ = checkRequiredGoVersion()
 
-	// Invoking build.go with no parameters at all is equivalent to "go run
-	// build.go install all" as that builds everything (incrementally),
+	// Invoking build.go with no parameters at all builds everything (incrementally),
 	// which is what you want for maximum error checking during development.
 	if flag.NArg() == 0 {
-		var tags []string
-		if noupgrade {
-			tags = []string{"noupgrade"}
```

Comment:
It seems so, but this diff is a bit poor. Look at line 178 (https://github.com/lkwg82/syncthing/blob/github-checks4/build.go#L178) there it is.